assets/app/features/upgrade/controllers/upgrade7-backup.controller.js

```diff
@@ -28,17 +29,38 @@ function() {
 
             upgradeBackupFactory.create()
                 .then(
+                    function (response) {
+                        downloadBackup(response.data);
+                    },
+                    // In case of backup error
+                    function (error) {
+                        vm.backup.error = error;
+                        vm.backup.running = false;
+                        vm.backup.completed = true;
+                    }
+                );
+        }
+
+        function extractFilename(headers) {
+            var value = headers['content-disposition'];
+            value = value.split(';')[1].trim().split('=')[1];
+            return value.replace(/"/g, '');
+        }
+
+        function downloadBackup(backupInfo) {
```

Comment:
Please, use backupId instead of backupInfo, there's no need to receive the full object